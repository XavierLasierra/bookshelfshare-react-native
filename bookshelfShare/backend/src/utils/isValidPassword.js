const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
