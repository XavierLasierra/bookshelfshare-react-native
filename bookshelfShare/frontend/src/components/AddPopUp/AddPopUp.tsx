import React from 'react';
import { Text } from 'react-native';
import {
  Menu,
  MenuOptions,
  MenuTrigger,
  renderers,
  MenuOption
} from 'react-native-popup-menu';
import { IAddPopUpProps } from '../../types/interfaces';

import BarCodeIcon from '../../assets/barCodeIcon.svg';
import SearchIcon from '../../assets/whiteSearchIcon.svg';
import styles from './addPopUp.styles';
import globalStyles from '../../styles/global.styles';

export default function AddPopUp({ navigation }: IAddPopUpProps) {
  function handleClickScan() {
    navigation.navigate({
      name: 'BarCodeScanner',
      merge: true
    });
  }

  function handleClickSearch() {
    navigation.navigate({
      name: 'BookSearchNavigator',
      merge: true
    });
  }

  return (
    <Menu
      style={styles.addButton}
      renderer={renderers.Popover}
    >
      <MenuTrigger>
        <Text style={styles.addButtonText}>+</Text>
      </MenuTrigger>
      <MenuOptions
        optionsContainerStyle={styles.popUp}
        testID="menuOptions"
      >
        <MenuOption
          key="scanPage"
          style={[globalStyles.button, styles.popUpButton]}
          onSelect={handleClickScan}
        >
          <BarCodeIcon style={styles.popUpIcon} width={30} height={30} />
          <Text style={globalStyles.buttonText}>Scan</Text>
        </MenuOption>
        <MenuOption
          key="searchPage"
          style={[globalStyles.button, styles.popUpButton]}
          onSelect={handleClickSearch}
        >
          <SearchIcon style={styles.popUpIcon} width={30} height={30} />
          <Text style={globalStyles.buttonText}>Search</Text>
        </MenuOption>
      </MenuOptions>
    </Menu>
  );
}
