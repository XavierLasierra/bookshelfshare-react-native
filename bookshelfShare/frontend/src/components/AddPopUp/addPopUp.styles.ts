import {StyleSheet} from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  addButton: {
    position: 'relative',
    top: -10,
    width: 60,
    height: 60,
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 2,
    borderRadius: 50,
    backgroundColor: stylesConstants.colors.white,
  },
  addButtonText: {
    fontSize: 30,
    color: stylesConstants.colors.mainText,
    width: 60,
    lineHeight: 60,
    height: 60,
    textAlign: 'center',
  },
  popUp: {
    backgroundColor: stylesConstants.colors.white,
    height: 220,
    width: 250,
    padding: 20,
    borderRadius: 20,
    justifyContent: 'space-between',
  },
  popUpButton: {
    marginVertical: 10,
  },
  popUpIcon: {
    marginRight: 20,
    position: 'absolute',
    left: 20,
  },
});
