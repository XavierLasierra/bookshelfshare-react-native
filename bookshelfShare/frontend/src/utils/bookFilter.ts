export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
