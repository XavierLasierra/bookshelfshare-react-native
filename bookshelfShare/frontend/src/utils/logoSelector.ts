import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
