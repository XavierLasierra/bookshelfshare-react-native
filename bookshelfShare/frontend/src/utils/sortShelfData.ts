export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
