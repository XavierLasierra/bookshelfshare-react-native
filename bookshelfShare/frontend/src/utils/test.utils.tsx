import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
