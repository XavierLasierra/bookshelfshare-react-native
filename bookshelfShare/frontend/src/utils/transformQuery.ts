interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
