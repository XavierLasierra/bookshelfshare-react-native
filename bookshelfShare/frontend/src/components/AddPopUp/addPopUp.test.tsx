import React from 'react';
import {MenuProvider} from 'react-native-popup-menu';
import AddPopUp from './AddPopUp';
import {render} from '../../utils/test.utils';
import navigationMock from '../../mocks/navigation.mock';

describe('Given an AddPopUp component', () => {
  describe('When it is rendered', () => {
    let screen: any;
    beforeEach(() => {
      screen = render(
        <MenuProvider>
          <AddPopUp navigation={navigationMock} />
        </MenuProvider>,
      );
    });
    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});
