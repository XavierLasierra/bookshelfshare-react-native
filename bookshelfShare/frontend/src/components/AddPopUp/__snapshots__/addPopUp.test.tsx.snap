// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an AddPopUp component When it is rendered Then should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 50,
          "elevation": 2,
          "height": 60,
          "justifyContent": "center",
          "position": "relative",
          "top": -10,
          "width": 60,
        }
      }
    >
      <View
        collapsable={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            ctx={
              Object {
                "menuActions": Object {
                  "_getOpenedMenu": [Function],
                  "_notify": [Function],
                  "closeMenu": [Function],
                  "isMenuOpen": [Function],
                  "openMenu": [Function],
                  "toggleMenu": [Function],
                },
                "menuRegistry": Object {
                  "getAll": [Function],
                  "getMenu": [Function],
                  "setOptionsCustomStyles": [Function],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                  "updateLayoutInfo": [Function],
                },
              }
            }
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#3E3C37",
                  "fontSize": 30,
                  "height": 60,
                  "lineHeight": 60,
                  "textAlign": "center",
                  "width": 60,
                }
              }
            >
              +
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
